api/context.go

```diff
@@ -463,3 +519,39 @@ func Handle404(w http.ResponseWriter, r *http.Request) {
 	l4g.Error("%v: code=404 ip=%v", r.URL.Path, GetIpAddress(r))
 	RenderWebError(err, w, r)
 }
+
+func CreateOAuthSession(token string) (*model.Session, *model.AppError) {
+	token = model.Md5Encrypt(utils.Cfg.ServiceSettings.TokenSalt, token)
```

Comment:
Needs to be encrypted in the database, but it also needs to encrypt to the same thing every time which is why I didn't use bcrypt (since that uses it's own salt). I used MD5 because it encrypts to a hexadecimal which will always be a database friendly string